tests/database/helpers/EventAccumulator.ts

```diff
@@ -0,0 +1,30 @@
+export class EventAccumulator {
+  public eventData = [];
+  public promise;
+  public resolve;
+  public reject;
+  constructor(private expectedEvents: number) {
+    if (!this.expectedEvents) throw new Error('EventAccumulator:You must pass a number of expected events to the constructor');
+    this.promise = new Promise((resolve, reject) => {
+      this.resolve = resolve;
+      this.reject = reject;
+    });
+  }
+  addEvent(eventData?: any) {
+    this.eventData = [
+      ...this.eventData,
+      eventData
+    ];
+    if (this.eventData.length >= this.expectedEvents) {
+      this.resolve(this.eventData);
```

Comment:
A promise can only resolve once, so if it gets hit multiple times, only the first invocation will actually trigger a call to `.then`. All other invocations are ignored.